Update swagger to openapi (#36)
ballerina-guides/open-api-based-service: README.md

[![Build Status](https://travis-ci.org/ballerina-guides/open-api-based-service.svg?branch=master)](https://travis-ci.org/ballerina-guides/open-api-based-service)

# OpenAPI

> [OpenAPI Specification](https://swagger.io/specification/) (formerly called the Swagger Specification) is a specification that creates RESTful contract for APIs, detailing all of its resources and operations in a human and machine-readable format for easy development, discovery, and integration.

Reviewer: ```suggestion
The [OpenAPI specification](https://swagger.io/specification/) (formerly called the Swagger specification) creates RESTful contracts for APIs. It details all resources and operations of APIs in a human and machine-readable format for easy development, discovery, and integration.
```

> The OpenApi to Ballerina Code Generator can take existing OpenApi definition files and generate Ballerina services from them.

Reviewer 2: ```suggestion
The OpenAPI to Ballerina Code Generator can take existing OpenAPI definition files and generate Ballerina services from them.
```

> The OpenAPI to Ballerina Code Generator can take existing OpenAPI definition files and generate Ballerina services from them.

Reviewer: ```suggestion
The OpenAPI to Ballerina Code Generator can generate Ballerina services from existing OpenAPI definition files.

```

- [Observability](#observability)

## What you'll build

> You'll build a RESTful web service using an OpenAPI specification. This guide will walk you through building a pet store server from an OpenAPI definition. The pet store service will have RESTful POST,PUT,GET and DELETE methods to handle pet data.

Reviewer: ```suggestion
You will build a RESTful web service using the OpenAPI specification. This guide will walk you through building a pet store server from an OpenAPI definition. The pet store service will have the RESTful POST, PUT, GET and DELETE methods to handle pet data.
```

> If you want to skip the basics, you can download the git repo and directly move to the "Testing" section by skipping  "Implementation" section.

### Understanding the OpenAPI specification

> The scenario that we use throughout this guide will base on [petstore.json](guide/petstore.json) OpenApi specification. The OpenAPI specification contains all the required details about the pet store RESTful API. Additionally, You can use the Api Designer view in Ballerina vscode extension to create and edit OpenAPI files. 

Reviewer 2: ```suggestion
The scenario that we use throughout this guide will be based on the [petstore.json](guide/petstore.json) OpenAPI specification. The OpenAPI specification contains all the required details about the pet store RESTful API. Additionally, You can use the API Designer view in Ballerina VSCode extension to create and edit OpenAPI files. 
```

> The scenario that we use throughout this guide will be based on the [petstore.json](guide/petstore.json) OpenAPI specification. The OpenAPI specification contains all the required details about the pet store RESTful API. Additionally, You can use the API Designer view in Ballerina VSCode extension to create and edit OpenAPI files. 

Reviewer: ```suggestion
The scenario used throughout this guide is based on the [petstore.json](guide/petstore.json) OpenAPI specification. The OpenAPI specification contains all the required details about the RESTful API of the pet store. Additionally, you can use the API Designer view in the Ballerina VSCode extension to create and edit the OpenAPI files. 
```

> #### Basic structure of petstore OpenAPI specification

Reviewer: ```suggestion
#### Basic structure of the pet store OpenAPI specification
```

}
```

> NOTE :  The above OpenAPI definition is only the basic structure. You can find the complete OpenAPI definition in [petstore.json](guide/petstore.json) file.

Reviewer 2: ```suggestion
NOTE :  The above OpenAPI definition is only the basic structure. You can find the complete OpenAPI definition in the [petstore.json](guide/petstore.json) file.
```

   $ ballerina init
```

### Genarate the web service from the OpenAPI definition

> Ballerina language is capable of understanding OpenAPI specifications. We can easily generate the web service just by typing the following command in the terminal.

Reviewer: ```suggestion
Ballerina language is capable of understanding OpenAPI specifications. Execute the below command in a Terminal, to generate the web service.
```

> Ballerina language is capable of understanding the OpenAPI specifications. We can easily generate the web service just by typing the following command structure in the terminal.

Reviewer 2: This sounds like it's repeating info from a previous section. @praneesha thoughts?

> Ballerina language is capable of understanding the OpenAPI specifications. We can easily generate the web service just by typing the following command structure in the terminal.

Reviewer 2: ```suggestion
Ballerina language is capable of understanding OpenAPI specifications. We can easily generate the web service just by typing the following command structure in the terminal.
```

```
ballerina openaapi gen-service <openApiFile> [-o <output directory name>] [-m <module name>] 
```

> For our pet store service we need to run the following command from the `/guide` in sample root directory(location where you have the petstore.json file) to generate the Ballerina service from the OpenAPI definition

Reviewer: ```suggestion
Navigate to the `/guide` directory in the sample root directory (i.e., the location where you have the `petstore.json` file) and execute the below command. This generates the Ballerina service for the pet store service from the OpenAPI definition.
```

```bash 
$ ballerina openapi gen-service petstore.json -m petstore
```

> The `-m` flag indicates the module name and `-o` flag indicates the file destination for the web service. These parameters are optional and can be used to have a customized module name and file location for the project.

Reviewer: ```suggestion
The `-m` flag denotes the module name and the `-o` flag indicates the destination of the file corresponding to the web service. These parameters are optional and can be used to have a customized module name and file location for the project.
```

### Implement the business logic for petstore 

> Now we have the Ballerina web service for the give `petstore.json` OpenApi file. Next task is to implement the business logic for functionality of each resource. The Ballerina OpenAPi generator has generated `ballerina_petstore_impl.bal` file inside the `open-api-based-service/guide/petstore`. We need to fill the `ballerina_petstore_impl.bal` as per the requirement. For simplicity, we will use an in-memory map to store the pet data. The following code is the completed pet store web service implementation. 

Reviewer 2: ```suggestion
Now we have the Ballerina web service for the given `petstore.json` OpenAPI file. Next task is to implement the business logic for functionality of each resource. The Ballerina OpenAPI generator has generated `ballerina_petstore_impl.bal` file inside the `open-api-based-service/guide/petstore` directory. We need to fill the `ballerina_petstore_impl.bal` as per the requirement. For simplicity, we will use an in-memory map to store the pet data. The following code is the completed pet store web service implementation. 
```

> Now we have the Ballerina web service for the given `petstore.json` OpenAPI file. Next task is to implement the business logic for functionality of each resource. The Ballerina OpenAPI generator has generated `ballerina_petstore_impl.bal` file inside the `open-api-based-service/guide/petstore` directory. We need to fill the `ballerina_petstore_impl.bal` as per the requirement. For simplicity, we will use an in-memory map to store the pet data. The following code is the completed pet store web service implementation. 

Reviewer: ```suggestion
After creating the Ballerina web service for the given `petstore.json` OpenAPI file, next, the business logic should be implemented for the functionality of each resource. 

To do this, fill the `open-api-based-service/guide/petstore/`ballerina_petstore_impl.bal` file generated by the Ballerina OpenAPI generator with the required details as preferred. 

For simplicity, this guide uses an in-memory map to store the pet data. The following code is the complete implementation of the pet store web service. 
```
